packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -77,20 +83,30 @@ function NavigationMenuSelector( {
 	const menuUnavailable =
 		hasResolvedNavigationMenus && currentMenuId === null;
 
-	useEffect( () => {
-		if ( ! hasResolvedNavigationMenus && ! canUserCreateNavigationMenu ) {
-			setSelectorLabel( __( 'Loading …' ) );
-		} else if ( noMenuSelected || noBlockMenus || menuUnavailable ) {
-			setSelectorLabel( __( 'Choose or create a Navigation menu' ) );
-		}
+	let selectorLabel = '';
 
+	if ( isCreatingMenu || isResolvingNavigationMenus ) {
+		selectorLabel = __( 'Loading …' );
+	} else if ( noMenuSelected || noBlockMenus || menuUnavailable ) {
+		// Note: classic Menus may be available.
+		selectorLabel = __( 'Choose or create a Navigation menu' );
+	} else {
+		// Current Menu's title.
+		selectorLabel = currentTitle;
+	}
+
+	useEffect( () => {
 		if (
 			isCreatingMenu &&
 			( createNavigationMenuIsSuccess || createNavigationMenuIsError )
 		) {
 			setIsCreatingMenu( false );
 		}
-	}, [ hasResolvedNavigationMenus, createNavigationMenuIsSuccess ] );
+	}, [
+		isCreatingMenu,
+		createNavigationMenuIsError,
+		createNavigationMenuIsSuccess,
```

Comment:
Should we also add `setIsCreatingMenu` as a dependency?

Reply:
`useState` setters are guaranteed by React to display referential equality. I think this is because they derive from `dispatch` which is the same. I can double check that but I'm pretty sure it's correct.

Reply:
I don't think we need to pass it as a dependency because we think it might change, only because we have a warning about undeclared dependencies.

Reply:
I committed a change. Does that work?